packages/beacon-node/src/chain/archiveStore/differentialState/stateSnapshot.ts

```diff
@@ -0,0 +1,88 @@
+import {ChainForkConfig} from "@lodestar/config";
+import {BeaconState, Slot} from "@lodestar/types";
+import {IBeaconDb} from "../../../db/interface.js";
+import {BeaconStateSnapshot} from "./ssz.js";
+
+/**
+ * @internal
+ */
+export function snapshotToBeaconState(
+  modules: {config: ChainForkConfig},
+  stateSnapshot: BeaconStateSnapshot
+): BeaconState {
+  const {config} = modules;
+  const target = config.getForkTypes(stateSnapshot.slot).BeaconState.deserialize(stateSnapshot.stateBytes);
+  const targetBalances = config.getForkTypes(stateSnapshot.slot).Balances.deserialize(stateSnapshot.balancesBytes);
+  target.balances = targetBalances;
+
+  if (target.slot !== stateSnapshot.slot) {
+    throw new Error(`Invalid slot in differential state: expected ${stateSnapshot.slot}, got ${target.slot}`);
+  }
+
+  return target;
+}
+
+/**
+ * @internal
+ */
+export function snapshotToBeaconStateBytes(
+  modules: {config: ChainForkConfig},
+  stateSnapshot: BeaconStateSnapshot
+): Uint8Array {
+  const {config} = modules;
+  const state = snapshotToBeaconState(modules, stateSnapshot);
+
+  return config.getForkTypes(state.slot).BeaconState.serialize(state);
```

Comment:
I think there is some way to combine `stateSnapshot.stateBytes` and `stateSnapshot.balanceBytes` by manipulating the bytes directly. That way we can avoid deserializing `stateBytes` and `balanceBytes`, and serializing the combination.

This can be a future improvement